### ISSUE TITLE
nodePath not used for test detection

### ISSUE THREAD

**Issue by dvirtz** [hbenl/vscode-mocha-test-adapter#144]
In settings I have
```
  "mochaExplorer.nodePath": "/usr/local/bin/node"
```
but when I do 
```
console.log(process.execPath);
```
from `.mocharc.js` it prints
```
[2021-02-17 13:29:35.410] [INFO] Worker (stdout): /Applications/Visual Studio Code.app/Contents/Frameworks/Code Helper (Renderer).app/Contents/MacOS/Code Helper (Renderer)
```
and only afterwards
```
[2021-02-17 13:29:35.644] [DEBUG] Using nodePath: /usr/local/bin/node
```

Seems like the setting is only read after tests are found.

<details>
<summary>full log panel</summary>
[2021-02-17 13:29:34.257] [INFO] Loading test files of /Users/dyitzchaki/github/reveal-compiler-explorer
[2021-02-17 13:29:34.260] [DEBUG] Using working directory: /Users/dyitzchaki/github/reveal-compiler-explorer
[2021-02-17 13:29:35.410] [INFO] Worker (stdout): /Applications/Visual Studio Code.app/Contents/Frameworks/Code Helper (Renderer).app/Contents/MacOS/Code Helper (Renderer)

[2021-02-17 13:29:35.422] [DEBUG] Using Mocha options: {"ui":"bdd","timeout":2000,"retries":0,"requires":[],"delay":false,"fullTrace":false,"exit":false,"asyncOnly":false,"parallel":false}
[2021-02-17 13:29:35.423] [DEBUG] Looking for test files ["./packages/reveal-test/test/consume.test.cjs"] in /Users/dyitzchaki/github/reveal-compiler-explorer
[2021-02-17 13:29:35.643] [DEBUG] Found test files ["/Users/dyitzchaki/github/reveal-compiler-explorer/packages/reveal-test/test/consume.test.cjs"]
[2021-02-17 13:29:35.644] [DEBUG] Using nodePath: /usr/local/bin/node
[2021-02-17 13:29:35.645] [DEBUG] Using environment variables from config: {}
[2021-02-17 13:29:35.647] [DEBUG] Spawning /Users/dyitzchaki/.vscode/extensions/hbenl.vscode-mocha-test-adapter-2.9.1/out/worker/bundle.js with IPC options {}
[2021-02-17 13:29:35.811] [INFO] Worker: Using the mocha package at /Users/dyitzchaki/github/reveal-compiler-explorer/node_modules/mocha
[2021-02-17 13:29:36.073] [INFO] Worker: Patching Mocha
[2021-02-17 13:29:36.131] [INFO] Worker: Loading files
[2021-02-17 13:29:36.142] [INFO] Worker: Trying to use Mocha's experimental ESM module loader
[2021-02-17 13:29:36.154] [INFO] Worker: Patching describe
[2021-02-17 13:29:36.155] [INFO] Worker: Patching describe.skip
[2021-02-17 13:29:36.155] [INFO] Worker: Patching describe.only
[2021-02-17 13:29:36.155] [INFO] Worker: Patching it
[2021-02-17 13:29:36.155] [INFO] Worker: Patching it.only
[2021-02-17 13:29:36.155] [INFO] Worker: Patching it.skip
[2021-02-17 13:29:36.156] [INFO] Worker: Copying it.retries
[2021-02-17 13:29:36.157] [INFO] Worker: Patching context
[2021-02-17 13:29:36.157] [INFO] Worker: Patching context.skip
[2021-02-17 13:29:36.157] [INFO] Worker: Patching context.only
[2021-02-17 13:29:36.157] [INFO] Worker: Patching specify
[2021-02-17 13:29:36.157] [INFO] Worker: Patching specify.only
[2021-02-17 13:29:36.157] [INFO] Worker: Patching specify.skip
[2021-02-17 13:29:36.157] [INFO] Worker: Copying specify.retries
[2021-02-17 13:29:36.157] [INFO] Worker: Patching describe
[2021-02-17 13:29:36.158] [INFO] Worker: Patching describe.skip
[2021-02-17 13:29:36.158] [INFO] Worker: Patching describe.only
[2021-02-17 13:29:36.158] [INFO] Worker: Patching it
[2021-02-17 13:29:36.158] [INFO] Worker: Patching it.only
[2021-02-17 13:29:36.158] [INFO] Worker: Patching it.skip
[2021-02-17 13:29:36.158] [INFO] Worker: Copying it.retries
[2021-02-17 13:29:36.159] [INFO] Worker: Patching context
[2021-02-17 13:29:36.159] [INFO] Worker: Patching context.skip
[2021-02-17 13:29:36.160] [INFO] Worker: Patching context.only
[2021-02-17 13:29:36.161] [INFO] Worker: Patching specify
[2021-02-17 13:29:36.162] [INFO] Worker: Patching specify.only
[2021-02-17 13:29:36.163] [INFO] Worker: Patching specify.skip
[2021-02-17 13:29:36.167] [INFO] Worker: Copying specify.retries
[2021-02-17 13:29:36.530] [INFO] Worker: Looking for /Users/dyitzchaki/github/reveal-compiler-explorer/packages/reveal-test/test/consume.test.cjs in Error
    at findCallLocation (/Users/dyitzchaki/.vscode/extensions/hbenl.vscode-mocha-test-adapter-2.9.1/out/worker/bundle.js:849:17)
    at /Users/dyitzchaki/.vscode/extensions/hbenl.vscode-mocha-test-adapter-2.9.1/out/worker/bundle.js:840:30
    at Suite.<anonymous> (/Users/dyitzchaki/github/reveal-compiler-explorer/packages/reveal-test/test/consume.test.cjs:5:3)
    at Object.create (/Users/dyitzchaki/github/reveal-compiler-explorer/node_modules/mocha/lib/interfaces/common.js:148:19)
    at context.describe.context.context (/Users/dyitzchaki/github/reveal-compiler-explorer/node_modules/mocha/lib/interfaces/bdd.js:42:27)
    at /Users/dyitzchaki/.vscode/extensions/hbenl.vscode-mocha-test-adapter-2.9.1/out/worker/bundle.js:838:37
    at Object.<anonymous> (/Users/dyitzchaki/github/reveal-compiler-explorer/packages/reveal-test/test/consume.test.cjs:4:1)
    at Module._compile (internal/modules/cjs/loader.js:1085:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1114:10)
    at Module.load (internal/modules/cjs/loader.js:950:32)
[2021-02-17 13:29:36.530] [INFO] Worker: Looking for /Users/dyitzchaki/github/reveal-compiler-explorer/packages/reveal-test/test/consume.test.cjs in Error
    at findCallLocation (/Users/dyitzchaki/.vscode/extensions/hbenl.vscode-mocha-test-adapter-2.9.1/out/worker/bundle.js:849:17)
    at /Users/dyitzchaki/.vscode/extensions/hbenl.vscode-mocha-test-adapter-2.9.1/out/worker/bundle.js:840:30
    at Object.<anonymous> (/Users/dyitzchaki/github/reveal-compiler-explorer/packages/reveal-test/test/consume.test.cjs:4:1)
    at Module._compile (internal/modules/cjs/loader.js:1085:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1114:10)
    at Module.load (internal/modules/cjs/loader.js:950:32)
    at Function.Module._load (internal/modules/cjs/loader.js:791:14)
    at Module.require (internal/modules/cjs/loader.js:974:19)
    at require (internal/modules/cjs/helpers.js:88:18)
    at Object.exports.requireOrImport (/Users/dyitzchaki/github/reveal-compiler-explorer/node_modules/mocha/lib/esm-utils.js:20:12)
[2021-02-17 13:29:36.536] [INFO] Worker (stdout): 

[2021-02-17 13:29:36.537] [INFO] Worker (stdout): 

[2021-02-17 13:29:36.537] [INFO] Worker (stdout):   0 passing (2ms)

[2021-02-17 13:29:36.537] [INFO] Worker (stdout): 

[2021-02-17 13:29:36.538] [INFO] Worker: Converting tests and suites
[2021-02-17 13:29:36.539] [INFO] Received tests from worker
[2021-02-17 13:29:36.555] [INFO] Worker finished with code 0 and signal null
</details>

**Comment by hbenl** [comment]
There are 3 different tasks that Mocha Test Explorer runs in child processes:
* reading the Mocha configuration file
* detecting tests
* running tests

The configured node path is only used for detecting and running tests, but not for reading the configuration file (because it shouldn't make a difference there) and this is where that log message is coming from.
Is there any reason why you'd need the configuration to be read using the configured node path?

**Comment by dvirtz** [comment]
The reason is that I'm using top level await, supported from node 14.8, in
some of my tests so I'm checking the node version in mocha's configuration
file and only include those tests if applicable.

**Comment by hbenl** [comment]
This is fixed now (version 2.10.2).
